plugins/woocommerce/src/Blocks/Domain/Services/AddressAutocomplete.php

```diff
@@ -0,0 +1,97 @@
+<?php
+declare( strict_types=1 );
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+/**
+ * Service class for managing address autocomplete providers.
+ */
+class AddressAutocomplete {
```

Comment:
We discussed this in our meetup @opr but we now also want to support maps block, even if we're not going to do that right now, we should give that space to ourself later on. This is an internal class, so we can rename it later but I think we should move from `AddressAutocomplete` to something more generic that includes both a map and address search. It's a safe assumption that any service that allows you to embed a map, will also provide address search, as searching for an address is a prerequisite for a map. Things like mapbox, Apple Maps, and Google Maps, all provide address search.

I think we can rename this to something like AddressProvider, and remove the mention of autocomplete or mention it alongside a map

Reply:
I thought about it, and I think the classes we want the users to register are `AddressProvider`s so it makes sense for our class to be called something like `AddressProviderService`. Updated 👌🏼